calligraphy/src/main/java/uk/co/chrisjenx/calligraphy/CalligraphyFactory.java

```diff
@@ -150,7 +150,29 @@ void onViewCreatedInternal(View view, final Context context, AttributeSet attrs)
         // Toolbar(Which underlies the ActionBar) for its children.
         if (CalligraphyUtils.canCheckForV7Toolbar() && view instanceof android.support.v7.widget.Toolbar) {
             final Toolbar toolbar = (Toolbar) view;
-            toolbar.getViewTreeObserver().addOnGlobalLayoutListener(new ToolbarLayoutListener(this, context, toolbar));
+            boolean hadTitle = toolbar.getTitle() != null;
```

Comment:
That's not true, the text is null when empty and set otherwise. Though it should probably be safe to use isEmpty.